xdsl/dialects/utils/format.py

```diff
@@ -1,6 +1,4 @@
-from abc import ABC
 from collections.abc import Iterable, Sequence
```

Comment:
I'm not sure if `format` is really the best name to characterise these helpers. Have you thought about further splitting them into printer utils and parsing utils?

Reply:
These tend to be quite coupled to each other, so it made sense to keep them together. I agree that the name isn't great, but I can't think of a better one... We could also go with the irdl and ir route and ban the sub files of the utils, and expose it all via `xdsl.dialects.utils`

Reply:
I'll try that quickly, will result in smaller diff and probably a better experience

Reply:
I'm not sure I understand the point of the irdl approach, just seems to cause loads of linting errors for me

Reply:
For some reason, the local development experience is worse than for clients, Pylance automatically inserts the correct import if you pip install it.